samples/samples/snippets.py

```diff
@@ -2796,6 +2796,33 @@ def directed_read_options(
     # [END spanner_directed_read]
 
 
+def set_custom_timeout_and_retry(instance_id, database_id):
+    """Executes a snapshot read with custom timeout and retry."""
+    # [START spanner_set_custom_timeout_and_retry]
+    from google.api_core.retry import Retry
+
+    # instance_id = "your-spanner-instance"
+    # database_id = "your-spanner-db-id"
+    spanner_client = spanner.Client()
+    instance = spanner_client.instance(instance_id)
+    database = instance.database(database_id)
+
+    retry = Retry(initial=5, maximum=100, multiplier=2, timeout=60)
```

Comment:
Add comments for each retry option.

Reply:
PTAL.

Reply:
These values should be inline with the documentation . As per the below documentation , we should use `multiplier` as `1.5` , `initial` as `500ms` , `maxretrydelay` as `16seconds` which i guess is the `timeout` value here. 

Also we should add a retry for `DEADLINE_EXCEEDED` error.

`
In the following sample, a timeout of 60 seconds is set for the given operation. If the operation takes longer than this timeout, the operation fails with a DEADLINE_EXCEEDED error.

If the operation fails with an UNAVAILABLE error code, for example, if there is a transient network problem, the operation is retried. The client waits for 500 ms before starting the first retry attempt. If the first retry fails, the client waits for 1.5 * 500 ms = 750 ms before starting the second retry. This retry delay continues to increase until the operation either succeeds or reaches the maximum retry delay of 16 seconds. The operation fails with a DEADLINE_EXCEEDED error if the total time that is spent on trying the operation exceeds the total timeout value of 60 seconds.
`

Reply:
@surbhigarg92 
Thanks for pointing this out. I have modified the sample to align with the documentation
https://cloud.google.com/spanner/docs/custom-timeout-and-retry